matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/pushrules/EventMatchCondition.kt

```diff
@@ -56,7 +56,12 @@ class EventMatchCondition(
             if (wordsOnly) {
                 value.caseInsensitiveFind(pattern)
             } else {
-                val modPattern = if (pattern.hasSpecialGlobChar()) pattern.simpleGlobToRegExp() else "*$pattern*".simpleGlobToRegExp()
+                val modPattern = if (pattern.hasSpecialGlobChar())
+                    // Regex.containsMatchIn() is way faster without leading and trailing
+                    // stars, that don't make any difference for the evaluation result
+                    pattern.removePrefix("*").removeSuffix("*").simpleGlobToRegExp()
```

Comment:
TIL! :100: 

not a blocker for the PR (assuming this is working as intended), this class would be a great candidate for some unit tests

Reply:
We have some [here](https://github.com/vector-im/element-android/blob/main/matrix-sdk-android/src/test/java/org/matrix/android/sdk/api/pushrules/PushRulesConditionTest.kt#L34)!